File: py/937.py
class Solution:
    def reorderLogFiles(self, logs):
        """
        :type logs: List[str]
        :rtype: List[str]
        """
        logs1 = []
        logs2 = []
        for log in logs:
            log2 = log.split(" ")
            logid = log2[0]
            log2 = " ".join(log2[1:])
            if log2[0].isnumeric():
                logs1.append([logid, log2])
            else:
                logs2.append([logid, log2])
        logs2 = sorted(logs2, key=lambda log: log[1])
        logs2.extend(logs1)
        return list(map(lambda log: " ".join(log), logs2))

s = Solution()
print(s.reorderLogFiles(["a1 9 2 3 1", "g1 act car", "zo4 4 7", "ab1 off key dog", "a8 act zoo"]))
print(["g1 act car","a8 act zoo","ab1 off key dog","a1 9 2 3 1","zo4 4 7"])

File: js/937.js
var isNumber = function (num) {
  return !isNaN(Number(num))
}
/**
 * @param {string[]} logs
 * @return {string[]}
 */
var reorderLogFiles = function (logs) {

  const logs1 = []
  const logs2 = []

  for (let i = 0; i < logs.length; i++) {
    let [id, ...log] = logs[i].split(' ');
    log = log.join(' ')

    if (isNumber(log[0])) {
      logs1.push([id, log])
    } else {
      logs2.push([id, log])
    }
  }

  return logs2
    .sort((a, b) => a[1].localeCompare(b[1]))
    .concat(logs1)
    .map(e => e.join(' '))
};

var assert = require('assert');
assert.deepEqual(reorderLogFiles(["a1 9 2 3 1", "g1 act car", "zo4 4 7", "ab1 off key dog", "a8 act zoo"]), ["g1 act car", "a8 act zoo", "ab1 off key dog", "a1 9 2 3 1", "zo4 4 7"])